### ISSUE TITLE
Enhancing the Look and Feel of Eclipse IDE Icons

### ISSUE THREAD

**Issue by Michael5601** [eclipse-platform/eclipse.platform.images#108]
Icons play a critical role in shaping the overall **Look and Feel** of the Eclipse IDE. Unfortunately, the current icons appear outdated and suffer from several issues that impact their usability, appearance, and consistency. This issue aims to open a discussion to summarize these problems and outline the requirements for a modern, cohesive set of icons.

Over the past few weeks, I’ve gained insights into these challenges while introducing a feature to rasterize SVGs at runtime within the Eclipse IDE. Through this process, I encountered several recurring issues, particularly related to the state of the SVGs themselves. These include:

#### 1. Problems in the State of the SVGs
- **Unnecessary Content**: Many SVGs contain empty layers, unnecessary elements, or items outside the view box, which contribute nothing to the rendered icon.
- **Inconsistent Contrast**: Foreground-background contrast is often achieved using white filters around certain areas. These filters vary between icons and render differently depending on the rasterizer.
- **Misalignment in View Box**: Icons are frequently not centered properly within the view box. For example, icons intended for 16×16px may not fill the available space, leading to inconsistent visual sizes.
- **Asymmetrical Elements**: Certain design elements (e.g., the plus in the star of the following example) are not symmetrically placed.  
  *Example: The following icon (org.eclipse.jdt.ui\icons\full\etool16\java_app.svg) shows points 1–4:*  
  <img src="https://github.com/user-attachments/assets/590c0c52-6276-4381-9f79-6a9a0ffd2460" width="117" height="102">
- **Text as Text**: Some icons include text that hasn’t been converted to paths, resulting in inconsistent rendering across different rasterizers.
- **Legacy SVG Format**: As noted in [this issue](https://github.com/eclipse-platform/eclipse.platform.images/issues/48), many icons are in an outdated SVG format. I don't know if this causes any problems beside a window opening in Inkscape.

#### 2. Broader Design and Usability Challenges
Beyond technical issues in the SVGs, I collected the following design flaws of the icons:

- **Overuse of Gradients**: A significant portion (58%) of all SVG elements are gradients. While gradients were once popular, they now make the icons appear outdated.
- **Limited Theming Support**: The many colors of the icons make it challenging to adapt them for different themes. 
- **Dark Mode Issues**: Some icons do not render well in dark mode or when displayed in a disabled state.
- **Spacing of the icons**: Although the icons generally follow a similar aesthetic, it seems not like there are rules regarding placement or alignment within the view box.
- **Overly Detailed Designs**: Icons sometimes include many details that are impossible to see at their typical display size, adding unnecessary complexity without value.
- **Unclear Symbolism**: Some icons fail to convey their intended function.

To solve these problems I would propose the idea of a new icon set to fully remake the existing icons.
I think this would be a possibility for contributors that are interested in graphics design rather than programming. 
As this is a big task it can be divided into icon packs for the different Eclipse IDE projects such as JDT, Platform, PDE etc.

**Comment by IamLRBA** [comment]
Hello, it will be a pleasure to contribute to this issue.

**Comment by IamLRBA** [comment]
I find this project incredibly exciting, as icons play a crucial role in shaping the user experience of the Eclipse IDE, and modernizing them has the potential to significantly enhance usability and aesthetics.

From what I understand, this initiative focuses on:

- Addressing technical inconsistencies in the current SVG icons, such as unnecessary content, misalignment, and outdated formats.
- Tackling broader design challenges, including improving contrast, aligning with modern UI trends, and ensuring compatibility with dark mode and various themes.
- Proposing a cohesive and modern icon set to elevate the overall look and feel of Eclipse while improving clarity and functionality.

I’m particularly drawn to the creative aspect of designing scalable, responsive, and minimalist vector graphics.

@Michael5601 could you please provide further details about the workflow, expectations, and timeline? Additionally, would this involve collaborating closely with the Eclipse UI team just like the other contributors?

Otherwise, looking forward to hearing your thoughts and potentially getting started on this impactful initiative!
Thanks!

**Comment by Michael5601** [comment]
As I am not an experienced Eclipse UI contributor I think at first it is important to hear some opinions in the community. Maybe you could help out here @BeckerWdf.

We need to divide the task of remaking icons (if it is wanted this way) into smaller tasks and provide a modern style guide or improve the already existing [UI Guidelines](https://www.eclipse.org/articles/Article-UI-Guidelines/Toc.html). 

With a style guide we can achieve consistency in the workflow of many collaborators and we could have a base for a discussion at first. "Good" looking Icons are subjective to the individual view but we need to get a consence about the goals and the style at first. 

I can also assist in the creation of icons as soon as my bachelor thesis is done in February 2025. Until then I will be part of this discussion and I am glad to help with my knowledge about graphics design. 

@IamLRBA maybe you want to attend the [Eclipse IDE ‐ Developers Community Call](https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/wiki/Eclipse-IDE-%E2%80%90-Developers-Community-Call#meeting-minutes) that takes place every second thursday. Tomorrow is the next meeting and I also plan to talk about my project that is related to icons/UI. New faces are very welcome :) and more contributors could get aware of this issue.

**Comment by Bananeweizen** [comment]
Please be aware that there are technical constraints, which contradict some of your assumptions AFAIK. E.g. icons are not centered _on purpose_ and they are sometimes aligned at the top, sometimes at the bottom etc. _on purpose_. See https://www.eclipse.org/articles/Article-UI-Guidelines/Contents.html#IconSpecsSize_VDG.
I'm not sure if any of these styleguide rules are part of the automated conversion, but I guess not. Otherwise one could of course have all SVGs centered, and then derive an offset PNG via some automation, and the offset would always be in the right direction.

**Comment by IamLRBA** [comment]
> As I am not an experienced Eclipse UI contributor I think at first it is important to hear some opinions in the community. Maybe you could help out here @BeckerWdf.
> 
> We need to divide the task of remaking icons (if it is wanted this way) into smaller tasks and provide a modern style guide or improve the already existing [UI Guidelines](https://www.eclipse.org/articles/Article-UI-Guidelines/Toc.html).
> 
> With a style guide we can achieve consistency in the workflow of many collaborators and we could have a base for a discussion at first. "Good" looking Icons are subjective to the individual view but we need to get a consence about the goals and the style at first.
> 
> I can also assist in the creation of icons as soon as my bachelor thesis is done in February 2025. Until then I will be part of this discussion and I am glad to help with my knowledge about graphics design.
> 
> @IamLRBA maybe you want to attend the [Eclipse IDE ‐ Developers Community Call](https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/wiki/Eclipse-IDE-%E2%80%90-Developers-Community-Call#meeting-minutes) that takes place every second thursday and is moderated by @HannesWell. Tomorrow is the next meeting and I also plan to talk about my project that is related to icons/UI. New faces are very welcome :) and more contributors could get aware of this issue.

Thanks @Michael5601 I'll be sure to attend tomorrow's meeting to get myself acquainted with the agenda.
Thanks again!

**Comment by BeckerWdf** [comment]
> I'm not sure if any of these styleguide rules are part of the automated conversion, but I guess not.

It's not.

**Comment by Michael5601** [comment]
I propose the following workflow for this project. If you like you can leave a comment with refinements. 

### Proposed Workflow and Initial Tasks for Developing a Comprehensive Style Guide

1. **Icon Design Experiment**  
Select three representative icons to serve as a basis for exploration. Each icon will be independently designed by at least two contributors in three distinct styles—resulting in a total of nine icons. @IamLRBA if you want we can do this step together. These designs should not involve collaboration or discussion to preserve individual creative interpretations. Each style should use only the color black to focus on the design itself. This ensures that the icon set can be used across different themes and allows for automated colorization in the future.

2. **Style Selection and Community Polling**  
Share the proposed styles with the community and conduct a poll to determine the most popular option. The style receiving the most votes will then be refined based on community feedback and constructive comments.

3. **Developing the Style Guide**  
Create a style guide based on the selected style and the feedback. The style guide will define design rules based on the discussions and the existing UI Guidelines. It will serve as a reference to ensure consistency across the icon set.

4. **Icon Creation and Quality Control**  
Once the style guide is created, contributors can begin creating icons individually. New contributors can also join the effort. It also doesn't matter if it will take a while to remake all icons as the style guide itself would be a big accomplishment and can be used as a base for the effort. To ensure high-quality outputs, each new icon will undergo a quality control process through a dedicated Pull Request. The PR should follow a specific template, including the new icon, and must verify that the icon meets the following criteria:  
   - Works well on different background colors.  
   - Is rasterizable with JSVG.  
   - Has a clean and optimized SVG structure.  
   - Is not too detailed.
   - Can be inverted without issues.  
   - Functions effectively at different sizes.  
   - Follows the rules in the style guide.  
This structured process minimizes the workload for quality controllers while maintaining consistency in the icon set.
All finished icons need to be saved in a similar structure as the current icons. Every icon needs to have the name of the icon it will replace in the future so it can be assigned to the intended place.

5. **Onboarding and Promotion**  
Create a welcome page with detailed instructions for contributors and advertise the initiative to attract new participants. Highlight the opportunity for contributors to gain experience in graphic design or frontend development while engaging with Eclipse. For instance, advertise the project on LinkedIn, highlighting that participants can showcase their icons in their portfolios for job interviews.  

In the community meeting at thursday we talked a bit about this initiative and I felt like the concerns were mostly about the size of this task. By focusing on the proposed workflow at first I think we can create a platform where more collaborators are interested in participating. Also I think that creating 400-500 icons would already be enough to give the most used pages in the Eclipse IDE a new look without replacing all 1500 or so unique icons.

**Comment by weisJ** [comment]
As I have originally created JSVG with this exact purpose (creating a scalable iconset for user interfaces) here are my two cents regarding this topic :)

To make creating cohesive I think it would be nice to have some color palette, which icons can refer to. In the readme of JSVG there is an example on how to dynamically change the color inside a `SVGDocument`. This way a single icon could be used for both dark and light mode, as the only thing that needs to be changed is the palette itself.

Also here is an idea I always wanted to see carried out with an icon set (which svg files lend themselves to relatively easily). Embed in the icon some information about
- where the baseline of the icon is relative to its bounds. This helps with visually aligning icons to text next to them.
- or moreover allow specifying a visual bounding box e.g. an icon could have some portion hanging off to the left while "visual left edge" is further in the middle. The UI could make use of this information to improve (in the mentioned case left edge) alignment across multiple rows of content. In [MiG Layout](http://www.miglayout.com) this concept is called "visual padding".

Of course both of these features would need further support on the UI side.

**Comment by IamLRBA** [comment]
> I propose the following workflow for this project. If you like you can leave a comment with refinements.
> ### Proposed Workflow and Initial Tasks for Developing a Comprehensive Style Guide
> 
>     1. **Icon Design Experiment**
>        Select three representative icons to serve as a basis for exploration. Each icon will be independently designed by at least two contributors in three distinct styles—resulting in a total of nine icons. @IamLRBA if you want we can do this step together. These designs should not involve collaboration or discussion to preserve individual creative interpretations. Each style should use only the color black to focus on the design itself. This ensures that the icon set can be used across different themes and allows for automated colorization in the future.
>
I'm okay with that, I'll start right away!

**Comment by Michael5601** [comment]
> I'm okay with that, I'll start right away!

Thats great, thank you! We need to decide which three representative icons we want to design so our result are comparable. 
I also want to make clear that this is no competition but a method to gather good comparable results. 

How about the following three icons:

- org.eclipse.pde.ui\icons\obj16\debug_exc.png:
![debug_exc](https://github.com/user-attachments/assets/0250dd47-656c-4480-87dc-9f0935aaba3f)
- org.eclipse.ui\icons\full\etool16\pin_editor.png:
![pin_editor](https://github.com/user-attachments/assets/fe5cfa60-c83f-4e5b-9365-9d2f4e87b7a9)
- org.eclipse.ui.ide\icons\full\etool16\search_src.png
![search_src](https://github.com/user-attachments/assets/eba38357-eab2-4dca-84f1-2c22bfc0229d)

And we both design all icons in 16x16px size in black color with white background. 

As I also have another project that I need to finish, it would be nice if we have time to upload our results on next week Sunday, 1st of December in this discussion.

I appreciate you effort and wish you a lot of fun designing the icons!

**Comment by IamLRBA** [comment]
> > I'm okay with that, I'll start right away!
> 
> Thats great, thank you! We need to decide which three representative icons we want to design so our results are comparable. I also want to make clear that this is no competition but a method to gather good comparable results.
> 
> How about the following three icons:
> 
>     * org.eclipse.pde.ui\icons\obj16\debug_exc.png:
>       ![debug_exc](https://private-user-images.githubusercontent.com/102024826/389454392-0250dd47-656c-4480-87dc-9f0935aaba3f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI1Mjc5ODUsIm5iZiI6MTczMjUyNzY4NSwicGF0aCI6Ii8xMDIwMjQ4MjYvMzg5NDU0MzkyLTAyNTBkZDQ3LTY1NmMtNDQ4MC04N2RjLTlmMDkzNWFhYmEzZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyNVQwOTQxMjVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iYTY3YjMxMGE5YmM4ZjZmNjFmYjRmNzIzZjlkOTZkNDVkNDY0OWU1ZmQyMzE1YzM4MGRiYjg2NmRjMGU1MWY1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.OZKR6VcozdVBe4DzYsMVX_ZVVEwyLH3UK3mlVN3Wtuc)
> 
>     * org.eclipse.ui\icons\full\etool16\pin_editor.png:
>       ![pin_editor](https://private-user-images.githubusercontent.com/102024826/389455632-fe5cfa60-c83f-4e5b-9365-9d2f4e87b7a9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI1Mjc5ODUsIm5iZiI6MTczMjUyNzY4NSwicGF0aCI6Ii8xMDIwMjQ4MjYvMzg5NDU1NjMyLWZlNWNmYTYwLWM4M2YtNGU1Yi05MzY1LTlkMmY0ZTg3YjdhOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyNVQwOTQxMjVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zY2NlMTBkNTYwMTVkZjJjZWQ3ODg2MzQ4MzIwZmY3MjIxZDA4YmYxYTY1ZGUzMmMzYWQ0YTk2YjViNzQ0OTM3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.LrCAt6sORHUR1Uv_VxsjUa2sibxuoVVVaQUXMRC32a8)
> 
>     * org.eclipse.ui.ide\icons\full\etool16\search_src.png
>       ![search_src](https://private-user-images.githubusercontent.com/102024826/389457340-eba38357-eab2-4dca-84f1-2c22bfc0229d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI1Mjc5ODUsIm5iZiI6MTczMjUyNzY4NSwicGF0aCI6Ii8xMDIwMjQ4MjYvMzg5NDU3MzQwLWViYTM4MzU3LWVhYjItNGRjYS04NGYxLTJjMjJiZmMwMjI5ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyNVQwOTQxMjVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hMTUwOGQ4M2I5OTBhOWMzZGEwMjhiYWM4Y2Q1ODg0MjFhMzMzMjYwZTM0ODQ4NzMzM2E4OWRmYjEwZjEzMjY1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.8cmJAxB_Lvs8zZTz0VmTX5AoKYvZdg141DrHdnyr38Q)
> 
> 
> And we both design all icons in 16x16px size in black color with white background.
> 
> As I also have another project that I need to finish, it would be nice if we have time to upload our results on next week Sunday, 1st of December in this discussion.
> 
> I appreciate your effort and wish you a lot of fun designing the icons!

Sure thing, Thanks brother and see you on the next Sunday.
If it's not any bother, could I get a means of directly communicating to you? That would be nice!

**Comment by Michael5601** [comment]
> > > I'm okay with that, I'll start right away!
> > 
> > 
> > Thats great, thank you! We need to decide which three representative icons we want to design so our results are comparable. I also want to make clear that this is no competition but a method to gather good comparable results.
> > How about the following three icons:
> > ```
> > * org.eclipse.pde.ui\icons\obj16\debug_exc.png:
> >   ![debug_exc](https://private-user-images.githubusercontent.com/102024826/389454392-0250dd47-656c-4480-87dc-9f0935aaba3f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI1Mjc5ODUsIm5iZiI6MTczMjUyNzY4NSwicGF0aCI6Ii8xMDIwMjQ4MjYvMzg5NDU0MzkyLTAyNTBkZDQ3LTY1NmMtNDQ4MC04N2RjLTlmMDkzNWFhYmEzZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyNVQwOTQxMjVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iYTY3YjMxMGE5YmM4ZjZmNjFmYjRmNzIzZjlkOTZkNDVkNDY0OWU1ZmQyMzE1YzM4MGRiYjg2NmRjMGU1MWY1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.OZKR6VcozdVBe4DzYsMVX_ZVVEwyLH3UK3mlVN3Wtuc)
> > 
> > * org.eclipse.ui\icons\full\etool16\pin_editor.png:
> >   ![pin_editor](https://private-user-images.githubusercontent.com/102024826/389455632-fe5cfa60-c83f-4e5b-9365-9d2f4e87b7a9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI1Mjc5ODUsIm5iZiI6MTczMjUyNzY4NSwicGF0aCI6Ii8xMDIwMjQ4MjYvMzg5NDU1NjMyLWZlNWNmYTYwLWM4M2YtNGU1Yi05MzY1LTlkMmY0ZTg3YjdhOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyNVQwOTQxMjVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zY2NlMTBkNTYwMTVkZjJjZWQ3ODg2MzQ4MzIwZmY3MjIxZDA4YmYxYTY1ZGUzMmMzYWQ0YTk2YjViNzQ0OTM3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.LrCAt6sORHUR1Uv_VxsjUa2sibxuoVVVaQUXMRC32a8)
> > 
> > * org.eclipse.ui.ide\icons\full\etool16\search_src.png
> >   ![search_src](https://private-user-images.githubusercontent.com/102024826/389457340-eba38357-eab2-4dca-84f1-2c22bfc0229d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI1Mjc5ODUsIm5iZiI6MTczMjUyNzY4NSwicGF0aCI6Ii8xMDIwMjQ4MjYvMzg5NDU3MzQwLWViYTM4MzU3LWVhYjItNGRjYS04NGYxLTJjMjJiZmMwMjI5ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyNVQwOTQxMjVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hMTUwOGQ4M2I5OTBhOWMzZGEwMjhiYWM4Y2Q1ODg0MjFhMzMzMjYwZTM0ODQ4NzMzM2E4OWRmYjEwZjEzMjY1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.8cmJAxB_Lvs8zZTz0VmTX5AoKYvZdg141DrHdnyr38Q)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > And we both design all icons in 16x16px size in black color with white background.
> > As I also have another project that I need to finish, it would be nice if we have time to upload our results on next week Sunday, 1st of December in this discussion.
> > I appreciate your effort and wish you a lot of fun designing the icons!
> 
> Sure thing, Thanks brother and see you on the next Sunday. If it's not any bother, could I get a means of directly communicating to you? That would be nice!

In my profile you can see my contact e-mail :)

**Comment by IamLRBA** [comment]
> > > > I'm okay with that, I'll start right away!
> > > 
> > > 
> > > Thats great, thank you! We need to decide which three representative icons we want to design so our results are comparable. I also want to make clear that this is no competition but a method to gather good comparable results.
> > > How about the following three icons:
> > > ```
> > > * org.eclipse.pde.ui\icons\obj16\debug_exc.png:
> > >   ![debug_exc](https://private-user-images.githubusercontent.com/102024826/389454392-0250dd47-656c-4480-87dc-9f0935aaba3f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI1Mjc5ODUsIm5iZiI6MTczMjUyNzY4NSwicGF0aCI6Ii8xMDIwMjQ4MjYvMzg5NDU0MzkyLTAyNTBkZDQ3LTY1NmMtNDQ4MC04N2RjLTlmMDkzNWFhYmEzZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyNVQwOTQxMjVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iYTY3YjMxMGE5YmM4ZjZmNjFmYjRmNzIzZjlkOTZkNDVkNDY0OWU1ZmQyMzE1YzM4MGRiYjg2NmRjMGU1MWY1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.OZKR6VcozdVBe4DzYsMVX_ZVVEwyLH3UK3mlVN3Wtuc)
> > > 
> > > * org.eclipse.ui\icons\full\etool16\pin_editor.png:
> > >   ![pin_editor](https://private-user-images.githubusercontent.com/102024826/389455632-fe5cfa60-c83f-4e5b-9365-9d2f4e87b7a9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI1Mjc5ODUsIm5iZiI6MTczMjUyNzY4NSwicGF0aCI6Ii8xMDIwMjQ4MjYvMzg5NDU1NjMyLWZlNWNmYTYwLWM4M2YtNGU1Yi05MzY1LTlkMmY0ZTg3YjdhOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyNVQwOTQxMjVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zY2NlMTBkNTYwMTVkZjJjZWQ3ODg2MzQ4MzIwZmY3MjIxZDA4YmYxYTY1ZGUzMmMzYWQ0YTk2YjViNzQ0OTM3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.LrCAt6sORHUR1Uv_VxsjUa2sibxuoVVVaQUXMRC32a8)
> > > 
> > > * org.eclipse.ui.ide\icons\full\etool16\search_src.png
> > >   ![search_src](https://private-user-images.githubusercontent.com/102024826/389457340-eba38357-eab2-4dca-84f1-2c22bfc0229d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI1Mjc5ODUsIm5iZiI6MTczMjUyNzY4NSwicGF0aCI6Ii8xMDIwMjQ4MjYvMzg5NDU3MzQwLWViYTM4MzU3LWVhYjItNGRjYS04NGYxLTJjMjJiZmMwMjI5ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyNVQwOTQxMjVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hMTUwOGQ4M2I5OTBhOWMzZGEwMjhiYWM4Y2Q1ODg0MjFhMzMzMjYwZTM0ODQ4NzMzM2E4OWRmYjEwZjEzMjY1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.8cmJAxB_Lvs8zZTz0VmTX5AoKYvZdg141DrHdnyr38Q)
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > And we both design all icons in 16x16px size in black color with white background.
> > > As I also have another project that I need to finish, it would be nice if we have time to upload our results on next week Sunday, 1st of December in this discussion.
> > > I appreciate your effort and wish you a lot of fun designing the icons!
> > 
> > 
> > Sure thing, Thanks brother and see you on the next Sunday. If it's not any bother, could I get a means of directly communicating to you? That would be nice!
> 
> In my profile you can see my contact e-mail :)

Alright! ✌🏻

**Comment by Michael5601** [comment]
> @Michael5601 @weisJ
> 
> > where the baseline of the icon is relative to its bounds.
> 
> https://eclipse-platform.github.io/ui-best-practices/#icon_size_placement
> 
> > I think it would be nice to have some color palette, which icons can refer to
> 
> https://eclipse-platform.github.io/ui-best-practices/#color_palette_themes
> 
> A lot of the UI system is described in that documentation, so please check it before starting. It would be a pity if new icons contradict some of the aspects described there. Improvements of that document can be made at https://github.com/eclipse-platform/ui-best-practices.

Thanks for the advice. This will be part of the creation process for the new style guide as described above. As @akurtakov metioned: If we find outdated infos in the existing UI Guidelines we will provide an issue/PR and discuss how we proceed with the information in the new style guide. For example: I don't think it is smart to create the icons without centering them in the middle. It seems to me that this workflow comes from technical debt as there is no process in the Eclipse UI to automatically apply an offset. Until all new icons are finished this technical debt is maybe fixed and we would need to change back many icons to a proper centered state.

**Comment by BeckerWdf** [comment]
> Please note that these best practices are up for update as a lot have changed since they were written.

But that does not mean that they are bad our outdated per se. I would prefer PRs for the existing style guide instead of a complete new one.

**Comment by vogella** [comment]
> The issue with changing this is that it will look strange when not all icons (e.g. object type icons in project explorer) are aligned in the same way (if some are centered, some are not).

I should start with the places in which the icons are currently not aligned, like the main toolbar. For example:

![image](https://github.com/user-attachments/assets/e8396121-168a-45f0-8e38-7d7c8e0d269c)

**Comment by vogella** [comment]
> So a PR to the "run" and the "external tools" icons would be the fix for that

Thanks would already be a good great step forward (I personally find these misaligned icons very bad looking). @BeckerWdf or @Michael5601 could you provide a PR to fix them?

**Comment by Michael5601** [comment]
> > So a PR to the "run" and the "external tools" icons would be the fix for that
> 
> Thanks would already be a good great step forward (I personally find these misaligned icons very bad looking). @BeckerWdf or @Michael5601 could you provide a PR to fix them?

I can create the PR but this would also mean that the size of the second and third icon needs to be changed. Is this wanted?

**Comment by Michael5601** [comment]
> > this would also mean that the size of the second and third icon needs to be changed. Is this wanted?
> 
> I would say yes. For the second one the green circle would must grow by one pixel to the left / down. For the third one the whole content would be moved down by one pixel. And the green circle would move to the right. The red box would not move right.

Please see this [PR](https://github.com/eclipse-platform/eclipse.platform.images/pull/111) for the fix.

**Comment by IamLRBA** [comment]
**Search**

| Number           | Style          | Zoomed           | 16x16px           | 32x32px          |

|---------------------|--------------------|--------------------|--------------------|--------------------|

| 0 | original | ![search_original_white_zoomed](https://github.com/user-attachments/assets/8e1c42d2-0860-427e-b2bb-cf3485e163c7) | ![search_original_16x16](https://github.com/user-attachments/assets/cc9aefe7-a50b-4521-8ff5-d94410116bf6) | ![search_original_32x32](https://github.com/user-attachments/assets/fb0ab767-959f-410c-9f6d-7faf242a1f4d) |

| 1 | style 1 (Outline) | ![search_icon_outline_100px](https://github.com/user-attachments/assets/716bf605-79c1-4ce4-b35f-a34dac620abd) | ![search_icon_outline_16px](https://github.com/user-attachments/assets/4c99a4e4-2384-4880-a708-6f5347db7d20) | 
![search_icon_outline_32px](https://github.com/user-attachments/assets/0fd9771f-c689-4afd-a4ac-e8fd9f606ee8)
 |

| 2 | (Filled) | 
![search_icon_filled_100px](https://github.com/user-attachments/assets/110df9d7-8178-4558-9328-642cd78da66a)
 | 
![search_icon_filled_16px](https://github.com/user-attachments/assets/76bbc322-e122-44fa-b102-9f863dd9d4a5)
 | 
![search_icon_filled_32px](https://github.com/user-attachments/assets/99dd5455-364c-44ac-bbc2-af858e59621e)
 |

| 3 | (Abstract) |  |  | 
![search_icon_abstract_32px](https://github.com/user-attachments/assets/0fb9826b-82ae-4ec9-b375-e8691460e0b9)
 |

 

**Debug_exc**

| Number           | Style          | Zoomed           | 16x16px           | 32x32px          |

|---------------------|--------------------|--------------------|--------------------|--------------------|

| 0 | original | ![debug_exc_original_white_zoomed](https://github.com/user-attachments/assets/0cd6d626-408f-457d-8e5f-0eb89cda5bc8) | ![debug_exc_original_white_16x16](https://github.com/user-attachments/assets/6ddbb658-aac1-45d8-9743-3c4f483ea41b) | ![debug_exc_original_white_32x32](https://github.com/user-attachments/assets/7da9cbf7-13ff-402f-99c9-c7e78ee780ad) |

| 1 | style 1 (Outline) | 
![debug_exc_icon_outline_100px](https://github.com/user-attachments/assets/b8603269-e2a9-49f2-85f0-06ff18988d43)
 | 
![debug_exc_icon_outline_16px](https://github.com/user-attachments/assets/15dff0ef-1d21-4072-b46f-ad3710348a05)
 | 
![debug_exc_icon_outline_32px](https://github.com/user-attachments/assets/cdf6df83-7db9-4910-9fee-3d71544862f5)
 |

| 2 | (Filled) | 
![debug_exc_icon_filled_100px](https://github.com/user-attachments/assets/98d50d97-9756-41e6-8cd4-5914b4331305)
 |  | 
![debug_exc_icon_filled_16px](https://github.com/user-attachments/assets/77a2049d-6167-4c6d-bad8-8413e2bd88dd)
 | 
![debug_exc_icon_filled_32px](https://github.com/user-attachments/assets/aba2690c-61a5-42a7-9f88-2b975dd5fce7)


| 3 | (Abstract) | ![debug_exc_icon_abstract_100px](https://github.com/user-attachments/assets/c69c150c-469b-4582-823e-051c8752ce93)
 | ![debug_exc_icon_abstract_16px](https://github.com/user-attachments/assets/7d700d51-b66b-44b2-837b-05759b6a2b25)
 | ![debug_exc_icon_abstract_32px](https://github.com/user-attachments/assets/39e193c6-ebe1-401b-b9d0-1b3b9f6b0fbf)
 |

 

**Pin_editor**

| Number           | Style          | Zoomed           | 16x16px           | 32x32px          |

|---------------------|--------------------|--------------------|--------------------|--------------------|

| 0 | original | ![pin_editor_original_white_zoomed](https://github.com/user-attachments/assets/8e89b647-30bd-471a-8812-35ee90f8091c) | ![pin_editor_original_white_16x16](https://github.com/user-attachments/assets/e559fce8-3ed3-4204-b21b-d210de162b36) | ![pin_editor_original_white_32x32](https://github.com/user-attachments/assets/86fec993-2650-44d7-8f93-f5561ebc1b2a) |

| 1 | style 1 (Outline) | ![Alt](https://via.placeholder.com/100) | ![Alt](https://via.placeholder.com/16) | ![Alt](https://via.placeholder.com/32) |

| 2 | (Filled) | ![Alt](https://via.placeholder.com/100) | ![Alt](https://via.placeholder.com/16) | ![Alt](https://via.placeholder.com/32) |

| 3 | (Abstract) | ![Alt](https://via.placeholder.com/100) | ![Alt](https://via.placeholder.com/16) | ![Alt](https://via.placeholder.com/32) |

![debug_exc_icon_filled_16px](https://github.com/user-attachments/assets/ab1d2bbb-a841-46b7-878f-acfdca2939c9)

**Comment by Michael5601** [comment]
The following table showcases the prototypes of the icon styles I designed. Please view them in light mode in your browser, as they were created in black. While they are visible in dark mode, the design is intended for light mode.

Initially, we decided to focus only on monochromatic black icons. However, I experimented with an additional dual-tone style that allows for colorization. As a result, I designed four different styles instead of three. As you will see some styles are not suitable for the samll icon size, this was a nice learning for me. 

@IamLRBA, I’m excited to see your designs! As planned, I’ll soon create a poll for the community to vote on their favorite style. The selected design can be discussed and improved afterwards. 

**Search**

| Number | style | zoomed | 16x16px | 32x32px |
|---------------------|--------------------|--------------------|--------------------|--------------------|
| 0 | original | ![search_original_white_zoomed](https://github.com/user-attachments/assets/8e1c42d2-0860-427e-b2bb-cf3485e163c7) | ![search_original_16x16](https://github.com/user-attachments/assets/cc9aefe7-a50b-4521-8ff5-d94410116bf6) | ![search_original_32x32](https://github.com/user-attachments/assets/fb0ab767-959f-410c-9f6d-7faf242a1f4d) |
| 1 | mono | ![search_style1_mono_white_zoomed](https://github.com/user-attachments/assets/635bd004-ae52-4b0c-8029-0dd7863e8660) | ![search_style1_mono_white_16x16](https://github.com/user-attachments/assets/9dc41596-0f5c-49e0-8536-366284f536ab) | ![search_style1_mono_white_32x32](https://github.com/user-attachments/assets/8f706950-8073-4d2c-93ab-39e96033be24) |
| 2 | dual-tone | ![search_style2_dual-tone_white_zoomed](https://github.com/user-attachments/assets/b80ade13-056f-4ac2-b3ed-0ff643d363b8) | ![search_style2_dual-tone_white_16x16](https://github.com/user-attachments/assets/49e03ca8-bcf3-4710-9027-20f4b5cd0974) | ![search_style2_dual-tone_white_32x32](https://github.com/user-attachments/assets/6e199759-5fd6-4614-8615-0c6bc22b15c0) |
| 3 | outline | ![search_style3_outline_white_zoomed](https://github.com/user-attachments/assets/3e50d568-f250-4ca2-850b-b67f377a75de) | ![search_style3_outline_white_16x16](https://github.com/user-attachments/assets/53a835e4-40ff-4065-9a99-0e7b6b0d2352) | ![search_style3_outline_white_32x32](https://github.com/user-attachments/assets/d6baa2e3-e991-4fd6-8d16-69c6e68ffe35) |
| 4 | one-line | ![search_style4_one-line_white_zoomed](https://github.com/user-attachments/assets/2e689149-2224-4d39-95e2-ca17597f37fd) | ![search_style4_one-line_white_16x16](https://github.com/user-attachments/assets/0ba8f4b3-e521-437e-a4ab-4969bd452c2a) | ![search_style4_one-line_white_32x32](https://github.com/user-attachments/assets/ed642229-9dc7-4119-bc89-16f10387322e) |

**Debug_exc**

| Number | style | zoomed | 16x16px | 32x32px |
|---------------------|--------------------|--------------------|--------------------|--------------------|
| 0 | original | ![debug_exc_original_white_zoomed](https://github.com/user-attachments/assets/0cd6d626-408f-457d-8e5f-0eb89cda5bc8) | ![debug_exc_original_white_16x16](https://github.com/user-attachments/assets/6ddbb658-aac1-45d8-9743-3c4f483ea41b) | ![debug_exc_original_white_32x32](https://github.com/user-attachments/assets/7da9cbf7-13ff-402f-99c9-c7e78ee780ad) |
| 1 | mono | ![debug_style1_mono_white_zoomed](https://github.com/user-attachments/assets/bd304451-72a1-48de-bf3e-d9ba018234ab) | ![debug_style1_mono_white_16x16](https://github.com/user-attachments/assets/86560d72-e51b-461b-9a22-fd8c88b5dbf7) | ![debug_style1_mono_white_32x32](https://github.com/user-attachments/assets/46a984ec-abd0-41cc-823d-a37f62d1ed6c) |
| 2 | dual-tone | ![debug_style2_dual-tone_white_zoomed](https://github.com/user-attachments/assets/3944b5ca-1533-4972-a388-f5a32a21afb2) | ![debug_style2_dual-tone_white_16x16](https://github.com/user-attachments/assets/8e6f4062-c4e9-45dc-8224-0d539140b700) | ![debug_style2_dual-tone_white_32x32](https://github.com/user-attachments/assets/19d858f7-5482-4085-ae00-bfc739489266) |
| 3 | outline | ![debug_style3_outline_white_zoomed](https://github.com/user-attachments/assets/b9a02399-84f5-414a-b93e-1c8d138b2b85) | ![debug_style3_outline_white_16x16](https://github.com/user-attachments/assets/269062ab-235a-4248-9225-b156f60e8df3) | ![debug_style3_outline_white_32x32](https://github.com/user-attachments/assets/6065bd1f-ca1b-4f1d-ba4d-b1743d93be9b) |
| 4 | one-line | ![debug_style4_one-line_white_zoomed](https://github.com/user-attachments/assets/3d6604d6-0647-4c3f-9c1c-b1082373a48e) | ![debug_style4_one-line_white_16x16](https://github.com/user-attachments/assets/86a45e88-3f6c-4368-85e7-eae492ab58a1) | ![debug_style4_one-line_white_32x32](https://github.com/user-attachments/assets/b4fc2e0c-c53e-4e31-9819-5206e1fad46e) |

**Pin_Editor**
| Number | style | zoomed | 16x16px | 32x32px |
|---------------------|--------------------|--------------------|--------------------|--------------------|
| 0 | original | ![pin_editor_original_white_zoomed](https://github.com/user-attachments/assets/8e89b647-30bd-471a-8812-35ee90f8091c) | ![pin_editor_original_white_16x16](https://github.com/user-attachments/assets/e559fce8-3ed3-4204-b21b-d210de162b36) | ![pin_editor_original_white_32x32](https://github.com/user-attachments/assets/86fec993-2650-44d7-8f93-f5561ebc1b2a) |
| 1 | mono | ![pin_editor_style1_mono_white_zoomed](https://github.com/user-attachments/assets/77fac690-fe68-461d-b550-1a7f3f5aa498) | ![pin_editor_style1_mono_white_16x16](https://github.com/user-attachments/assets/4c03a0e1-7045-4f2a-a1b1-1276b6de859f) | ![pin_editor_style1_mono_white_32x32](https://github.com/user-attachments/assets/62597d92-ce7c-403d-b1f0-250b8ca93d7f) |
| 2 | dual-tone | ![pin_editor_style2_dual-tone_white_zoomed](https://github.com/user-attachments/assets/945fbce2-2440-4989-b464-cba8e3cc4c26) | ![pin_editor_style2_dual-tone_white_16x16](https://github.com/user-attachments/assets/fc21ba50-1ce9-45a3-8ed8-2038c92d4d23) | ![pin_editor_style2_dual-tone_white_32x32](https://github.com/user-attachments/assets/b55a22b9-ec68-419b-94db-116bb9aa1468) |
| 3 | outline | ![pin_editor_style3_outline_white_zoomed](https://github.com/user-attachments/assets/450a8fae-d895-45a4-a1a9-df5bcd10b7d3) | ![pin_editor_style3_outline_white_16x16](https://github.com/user-attachments/assets/0a0348c7-8795-45d7-a570-8a6b902caed5) | ![pin_editor_style3_outline_white_32x32](https://github.com/user-attachments/assets/8ba6a270-b7c9-4796-8e1d-e392f7423322) |
| 4 | one-line | ![pin_editor_style4_one-line_white_100x100](https://github.com/user-attachments/assets/9798f6de-ea10-4e74-9500-b25c9306721f) | ![pin_editor_style4_one-line_white_16x16](https://github.com/user-attachments/assets/a9927568-79ef-40be-aa8f-f35950d1c6be) | ![pin_editor_style4_one-line_white_32x32](https://github.com/user-attachments/assets/41dee2ef-bffa-4ad2-ac7b-e40181fa6c92) |

**Comment by IamLRBA** [comment]
Thanks @Michael5601, Here's my approach
 **Search**

| Number           | Style          | Zoomed           | 16x16px           | 32x32px          |

|---------------------|--------------------|--------------------|--------------------|--------------------|

| 0 | original | ![search_original_white_zoomed](https://github.com/user-attachments/assets/8e1c42d2-0860-427e-b2bb-cf3485e163c7) | ![search_original_16x16](https://github.com/user-attachments/assets/cc9aefe7-a50b-4521-8ff5-d94410116bf6) | ![search_original_32x32](https://github.com/user-attachments/assets/fb0ab767-959f-410c-9f6d-7faf242a1f4d) |

| 1 | Outline | ![search_icon_outline_100px](https://github.com/user-attachments/assets/aa498307-6278-4224-b456-c01c6a0676be) | ![search_icon_outline_16px](https://github.com/user-attachments/assets/e940025f-e64a-4353-892e-188c6198f993) | ![search_icon_outline_32px](https://github.com/user-attachments/assets/3b6bcc2d-a6b7-495e-a16d-5c5f369426e2) |

| 2 | Solid | ![search_icon_filled_100px](https://github.com/user-attachments/assets/6f15b69d-903b-4293-b983-5bdb7a0efce3) | ![search_icon_filled_16px](https://github.com/user-attachments/assets/2c013bab-59f0-4488-b436-dba36f775de6) | ![search_icon_filled_32px](https://github.com/user-attachments/assets/df185a7f-f086-4bdb-a63d-aeb51b600755) |

| 3 | Abstract | ![search_icon_abstract_100px](https://github.com/user-attachments/assets/4c33bdd2-cbb2-40c3-8c59-3c81188c39a4) | ![search_icon_abstract_16px](https://github.com/user-attachments/assets/e36fdf19-2fa3-42a4-93ac-78edf97f77b1) | ![search_icon_abstract_32px](https://github.com/user-attachments/assets/b21101b8-51e9-4e81-83af-aecb74a5fafe) |

 

**Debug_exc**

| Number           | Style          | Zoomed           | 16x16px           | 32x32px          |

|---------------------|--------------------|--------------------|--------------------|--------------------|

| 0 | original | ![debug_exc_original_white_zoomed](https://github.com/user-attachments/assets/0cd6d626-408f-457d-8e5f-0eb89cda5bc8) | ![debug_exc_original_white_16x16](https://github.com/user-attachments/assets/6ddbb658-aac1-45d8-9743-3c4f483ea41b) | ![debug_exc_original_white_32x32](https://github.com/user-attachments/assets/7da9cbf7-13ff-402f-99c9-c7e78ee780ad) |

| 1 | Outline | ![debug_exc_icon_outline_100px](https://github.com/user-attachments/assets/9dc5a69c-409c-4d57-b18a-5d9b9e7b0f65) | ![debug_exc_icon_outline_16px](https://github.com/user-attachments/assets/f632ab2e-3467-4159-a9c8-2ae622ca9440) | ![debug_exc_icon_outline_32px](https://github.com/user-attachments/assets/9d9a55a0-f85f-4a66-bc64-f0c8be33a4ed) |

| 2 | Solid | ![debug_exc_icon_filled_100px](https://github.com/user-attachments/assets/1b861439-b001-4756-996b-96468d9571db) | ![debug_exc_icon_filled_16px](https://github.com/user-attachments/assets/97b8f38c-daf0-4808-9b5f-ce048fde92ac) | ![debug_exc_icon_filled_32px](https://github.com/user-attachments/assets/a347bf04-923f-44c2-aae7-4540840868ea) |

| 3 | Abstract |![debug_exc_icon_abstract_100px](https://github.com/user-attachments/assets/28bc32bf-9b87-4860-98ca-ba1622ff5493) | ![debug_exc_icon_abstract_16px](https://github.com/user-attachments/assets/9fe46ad6-8318-4148-b7a0-36975d202222) | ![debug_exc_icon_abstract_32px](https://github.com/user-attachments/assets/34902e6f-631b-418c-a09a-306570217f12) |

 

**Pin_editor**

| Number           | Style          | Zoomed           | 16x16px           | 32x32px          |

|---------------------|--------------------|--------------------|--------------------|--------------------|

| 0 | original | ![pin_editor_original_white_zoomed](https://github.com/user-attachments/assets/8e89b647-30bd-471a-8812-35ee90f8091c) | ![pin_editor_original_white_16x16](https://github.com/user-attachments/assets/e559fce8-3ed3-4204-b21b-d210de162b36) | ![pin_editor_original_white_32x32](https://github.com/user-attachments/assets/86fec993-2650-44d7-8f93-f5561ebc1b2a) |

| 1 | Outline | ![pin_editor_icon_outline_100px](https://github.com/user-attachments/assets/37e52443-f20b-4db5-b07c-ab6ad96af525) | ![pin_editor_icon_outline_16px](https://github.com/user-attachments/assets/9b470e5d-05b6-43ee-89fb-b9f8a9128d30) | ![pin_editor_icon_outline_32px](https://github.com/user-attachments/assets/7467e25d-9c11-4bd2-a49a-fce5e32f0cde) |

| 2 | Solid | ![pin_editor_icon_filled_100px](https://github.com/user-attachments/assets/6b00abb0-7e44-47f7-8ce3-2ef5a64828d3) | ![pin_editor_icon_filled_16px](https://github.com/user-attachments/assets/5c0ea2d8-2af9-4303-86a9-0b63e23250db) | ![pin_editor_icon_filled_32px](https://github.com/user-attachments/assets/94540d15-0f34-4661-8878-61c4dde7dab2) |

| 3 | Abstract | ![pin_editor_icon_abstract_100px](https://github.com/user-attachments/assets/9a07c978-265e-4806-bc14-bdb63a9d73b8) | ![pin_editor_icon_abstract_16px](https://github.com/user-attachments/assets/8ce63669-fe8c-46d2-94cc-8ef3a78a6a19) | ![pin_editor_icon_abstract_32px](https://github.com/user-attachments/assets/e0d82437-4a9d-439b-a628-c7e48f738f8e) |

**Comment by IamLRBA** [comment]
Sorry my presentation is not appearing well, I'm trying to edit the markdown perhaps I wrote it wrong. 
For my designs I leaned into the Outline, Solid and Abstract art styles for their Simple, Bold and Modern values respectively. 
As it is my first time creating icons, it was a discovery and exploratory process for me and clearly there is too much room to improve. I tried to follow up with the UI guidelines suggested previously but failed to meet some (Especially regarding fitting the icon on the entire space of the BG).
Thanks for the opportunity @Michael5601 and I have learned a lot.

**Comment by Michael5601** [comment]
> Sorry my presentation is not appearing well, I'm trying to edit the markdown perhaps I wrote it wrong. For my designs I leaned into the Outline, Solid and Abstract art styles for their Simple, Bold and Modern values respectively. As it is my first time creating icons, it was a discovery and exploratory process for me and clearly there is too much room to improve. I tried to follow up with the UI guidelines suggested previously but failed to meet some (Especially regarding fitting the icon on the entire space of the BG). Thanks for the opportunity @Michael5601 and I have learned a lot.

No worries, your icons look great. Especially for your first icons :)
As these icons are only prototypes for the style that is gonna be chosen, it is not important to meet any criteria from the UI Guidelines. Thank you for participating in this important step.

I try to create the poll as soon as possible next week. I will rename the styles as we have 2 styles with the name outline right now, just so you know.

**Comment by vogella** [comment]
At the moment we cannot have icons set for light and dark theme as we do not support that in the framework. Is it planned to implement dark / light icon support?

**Comment by IamLRBA** [comment]
> 
> I can already see some things we can learn from these styles. We both interpreted the `search` function as a magnifying glass and we both had the idea to simplify the `pin_editor` icon to only the pin. I wanted to create only the pin for all styles at first but then decided against it to show how my styles would work for a complex design. I learned that the one-line style was not possible in a complex design which is why I wouldn't choose it for eclipse. This first challenge shows us many points that need to be discussed for the style guide.
> 

I personally liked your outline art style and thought it looked really nice (Especially with the pin_editor and debug_exc icons). It gives definition to the style. 
Also I'm impressed by how you maintained the board in the pin_editor icon. I initially thought I'd try it out but felt it would be overwhelming but it's awesome how you managed to pull it off and the idea is still readable. 
Cheers!

**Comment by Michael5601** [comment]
> At the moment we cannot have icons set for light and dark theme as we do not support that in the framework. Is it planned to implement dark / light icon support?

For the icon pack that we try to develop an icon theming support is needed. @akurtakov made me aware that someone tried to implement this but stopped working on the [project](https://github.com/eclipse-platform/eclipse.platform.ui/pull/14). This project needs to be finished but as it will take some time to develop this icon pack, I don't see a timing issue here. In the same instance a dark/light theme support for icons can be implemented. 

Also as @weisJ stated his library JSVG (that will probably soon be implemented in Eclipse for vector graphic support in my [project](https://github.com/eclipse-platform/eclipse.platform.swt/issues/1438)) allows rasterization and coloring of icons at runtime. This will be especially great for monochromatic/dual-tone icons as planned in this icon pack. Maybe we can implement the support for light/dark mode icons by just passing a flag in the rasterization process.

**Comment by Michael5601** [comment]
> > I can already see some things we can learn from these styles. We both interpreted the `search` function as a magnifying glass and we both had the idea to simplify the `pin_editor` icon to only the pin. I wanted to create only the pin for all styles at first but then decided against it to show how my styles would work for a complex design. I learned that the one-line style was not possible in a complex design which is why I wouldn't choose it for eclipse. This first challenge shows us many points that need to be discussed for the style guide.
> 
> I personally liked your outline art style and thought it looked really nice (Especially with the pin_editor and debug_exc icons). It gives definition to the style. Also I'm impressed by how you maintained the board in the pin_editor icon. I initially thought I'd try it out but felt it would be overwhelming but it's awesome how you managed to pull it off and the idea is still readable. Cheers!

Thank you :) I think my outline style is cool but not suitable for small detailed icons as the lines are too thin. It will be hard to simplify every eclipse icon and sometimes we need more details for the icon to work. Your outline style has thicker lines which works better. We will see what the community votes for.

**Comment by BeckerWdf** [comment]
It's really nice to see these proposals. Impressive work.
I have some remarks to provide:

It's really hard to evaluate icon proposals stand alone. Seeing them "in action" in the real UI helps a lot.

On OCX one participant saids: "I like the efforts modernize the UI. But pls. don't turn Eclipse visually into VSCode".
Maybe these icon proposals are "too different" to todays icons for a lot of users. What about provided a set of icons that is a simplification of todays version in the sense of that we remove radients but keep the colors but wit less saturation so that they look less colorful and bright.

When designing monochrome icons keep in mind that with that you will run into issues with the "disabled" icon-style Eclipse provides. See my the issues I have in https://github.com/eclipse-platform/eclipse.platform.images/issues/114#issuecomment-2507822852. You have to come up with an idea how you want to visualize a disabled state of an icon. Is that something VSCode even does offer?

> I think my outline style is cool but not suitable for small detailed icons as the lines are too thin.
Look at my experienced I talked about in:
https://speakerdeck.com/beckerwdf/modernize-the-ui-of-your-eclipse-application?slide=11
Lines that are thinner then 1px will be blurry and hard to see if rendered to 16x16px. So ideally lines should always be 1px.

**Comment by Michael5601** [comment]
Thank you for your valuable feedback. Here are my statements to your points:
> It's really hard to evaluate icon proposals stand alone. Seeing them "in action" in the real UI helps a lot.
> 
Maybe I can place them into a "real" environment for the poll. 

> What about provided a set of icons that is a simplification of todays version in the sense of that we remove radients but keep the colors but wit less saturation so that they look less colorful and bright.
> 
As soon as the icon theming feature is implemented, the users can decide which icons they want to use. The only discussion would be for the "default" icons but I don't mind if they stay in the current design as long as the user can switch to a modern style. Right now there is no such option and also no modern icon pack. 
Also I think Vector could already provide icons that are based on the current icons but in a flatened style. For this the icon theming feature is also needed.

> When designing monochrome icons keep in mind that with that you will run into issues with the "disabled" icon-style Eclipse provides. You have to come up with an idea how you want to visualize a disabled state of an icon. Is that something VSCode even does offer?
> 
I already thought of this and I don't know of VSCode but IntelliJ also has monochromatic disabled icons as can be seen in the following screenshot. Disabled icons can be created very easily by darkening the gray tone in dark mode and light mode. They use a monochromatic style where many icons are standard gray and sometimes they are colored like the `run` icon. Sometimes they also use a second color in a dual-tone icon where the base is standard gray and the other color gives some information as the `Run with coverage` icon. We don't need to do it like them but I see it as inspiration for our new modern icon pack. 
![image](https://github.com/user-attachments/assets/922d4ba3-6d0b-4608-89f0-33312f48f52d)

**Comment by BeckerWdf** [comment]
> I already thought of this and I don't know of VSCode but IntelliJ also has monochromatic disabled icons as can be seen in the following screenshot. Disabled icons can be created very easily by darkening the gray tone in dark mode and light mode.

If the difference is between light and dark grey I think this will be a accessibility issue. If the difference is between an monochrom ikon (with a color different then grey) and grey that would be much better.

**Comment by Michael5601** [comment]
> If the difference is between light and dark grey I think this will be a accessibility issue. If the difference is between an monochrom icon (with a color different then grey) and a grey version of that icon that would be much better.
> 
I see your point but I must disagree here as it also works in other programs. 

> So in the IntelliJ screenshot all the icons have color so I can see that e.g. "Stop" id disabled. But the the "Import Tests from File..." entry is grey in the enabled state to. So yes we can see the difference but it's not as easy to see it.
> 
Most icons in IntelliJ are colored with the standard gray. In dark mode disabled icons are always in darker gray regardless of the original color of the icon. 

> Btw. See in the IntelliJ screenshot, that the complete menu entry is also greyed out in the same color. So this also helps. But this only helps in menus - in toolbars you often only have the icon.

The toolbar in IntelliJ is the same color as the menu. This is not the case in Eclipse as in dark mode the menus are darker than the toolbar. But I don't see the problem with disabling icons in a monochromatic design in Eclipse as we right now also use a darker gray tone in the menus for disabled icons than in the toolbar:
![image](https://github.com/user-attachments/assets/0cc48011-62c8-4106-a2ba-316d20ca088a)

**Comment by vogella** [comment]
> The toolbar in IntelliJ is the same color as the menu. 

I like this design, can you open a new issue for this, suggesting that we use the same color in the toolbar and menu for the dark theme? The team around @thomasritter  @BeckerWdf and @mvm-sap may want to evaluate this. It definitely would make the usage of new icons more consistent as we only would have to care about one common background color.

**Comment by Michael5601** [comment]
@IamLRBA can you please send me your icons as SVG so I can finish the poll? :)

**Comment by IamLRBA** [comment]
> @IamLRBA can you please send me your icons as SVG so I can finish the poll? :)

Alright, should I send on email or right here?

**Comment by Michael5601** [comment]
> > @IamLRBA can you please send me your icons as SVG so I can finish the poll? :)
> 
> Alright, should I send on email or right here?

Send me an e-mail please. Thank you!

**Comment by IamLRBA** [comment]
> 
> Send me an e-mail please. Thank you!

Coming right up!

**Comment by Michael5601** [comment]
I created the Community Poll [here](https://github.com/eclipse-platform/eclipse.platform.ui/discussions/2588)

It would be great if you could participate.
Please mind that I changed the styles `Abstract` and `Outline 1` as the lines were too thin. Now they look a lot better.
@BeckerWdf Thank you for the hint with the Eclipse Background. In the poll I put all icons on the real Eclipse Light- and Dark-Mode background colors so the icons can be seen in the right environment. 

Please ignore the voting options down here as they are automatically generated by the link in this message.

**Comment by zleonov** [comment]
Hi, I found this issue when looking for an issue about DevStyle. Have you seen their theme and icons? It might be useful to you guys.

![Image](https://github.com/user-attachments/assets/b01b3339-16ba-499c-be0e-336b29a8a4bb)

![Image](https://github.com/user-attachments/assets/f05a0cda-c201-4cb6-9110-3b34aab8595a)

**Comment by Michael5601** [comment]
> Hi, I found this issue when looking for an issue about DevStyle. I have long thought eclipse icons need a refresh, especially when using an HDPI resolution. But did not have time to really dig into the issue. Have you seen DevStyle Darkest Dark theme and icons? It might be useful to you guys (I customized the icons a bit for my purposes with their icon designer).
> 
> They use material fonts amongst others as the base for their icons.
> 
> ![Image](https://github.com/user-attachments/assets/b01b3339-16ba-499c-be0e-336b29a8a4bb)
> 
> ![Image](https://github.com/user-attachments/assets/f05a0cda-c201-4cb6-9110-3b34aab8595a)

Hi, thank you for the input. :)
I think the Darkest Dark theme was created by @BeckerWdf who already participates in the discussion. The theme looks great and maybe we can take some fitting icons also for our icon pack. In the long run I think the icons from the darkest dark theme should be a separate icon pack that can be chosen as soon as the technical part is implemented. 
I also don't know how many icons are part of the darkest dark theme.

I created a style guide for our modern (dual-tone) icon theme and right now I am trying to create a space where all new icons can be saved, so we can really start creating new icons.

**Comment by BeckerWdf** [comment]
> I think the Darkest Dark theme was created by [@BeckerWdf](https://github.com/BeckerWdf) who already participates in the discussion.

Noooo. It's a commercial plugin and I don't have anything to do with that.  Don't know the licence of that plugin and the icons - so be careful here.

**Comment by Michael5601** [comment]
> 
> Noooo. It's a commercial plugin and I don't have anything to do with that. Don't know the licence of that plugin and the icons - so be careful here.

Oh I'm sorry, then I mixed something up here.
